packages/connectivity/connectivity/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityBroadcastReceiver.java

```diff
@@ -49,8 +50,8 @@ public void onListen(Object arguments, EventChannel.EventSink events) {
 
   @Override
   public void onCancel(Object arguments) {
-    if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-      connectivity.getConnectivityManager().unregisterNetworkCallback(getNetworkCallback());
+    if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && networkCallback != null) {
+      connectivity.getConnectivityManager().unregisterNetworkCallback(networkCallback);
     } else {
       context.unregisterReceiver(this);
```

Comment:
We can now fall into this branch where we didn't before. That seems like a bug.

Reply:
And I can't implement this interface because require API level 21 : [ConnectivityManager.NetworkCallback](https://developer.android.com/reference/android/net/ConnectivityManager.NetworkCallback)
> So connectivity expects us to give it the same exact object for the callback when registering/unregistering?

Reply:
Right - it seems like if that's the case, we should just be doing nothing rather than calling `context.unregisterReceiver`